### PR TITLE
Docs: added documentation on how to configure storage retention

### LINKED ISSUE #1700: Document how to configure storage retention
The Mimir documentation doesn't explain how to configure storage retention.

We should document:
- How `-compactor.blocks-retention-period` works: explain by default is 0 that means retention is disabled (blocks are never deleted), explain it can be overridden on a per-tenant basis.
- Per-series deletion (including [this Prometheus API](https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series)) is currently not supported

Notes:
- We should create a dedicated page about the retention and link it from the compactor (the compactor page mentions "Deleting blocks that are no longer within a configurable retention period" but doesn't say anything else about it)
- We should make sure a search of "retention" in https://grafana.com/docs/mimir/latest finds the new retention doc page

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
In this PR:
- Documented how to configure the storage retention
- Re-organized a bit the ordering of configuration subpages, to show the most important first (storage and hash ring config)

#### Which issue(s) this PR fixes or relates to

Fixes #1700

#### Checklist

- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/operators-guide/architecture/components/compactor/index.md [hunk 1]
```diff
@@ -115,6 +115,13 @@ Soft deletion is based on a small `deletion-mark.json` file stored within the bl
 
 The soft delete mechanism gives time to queriers, rulers, and store-gateways to discover the new compacted blocks before the original blocks are deleted. If those original blocks were immediately hard deleted, some queries involving the compacted blocks could temporarily fail or return partial results.
 
+## Blocks retention
+
+The compactor is responsible to enforce the storage retention, deleting from the long-term storage the blocks containing samples older than the configured retention period.
+The storage retention is disabled by default, and no data will be deleted from the long-term storage unless you explicitly configure the retention period.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/architecture/components/compactor/index.md:121]:
```suggestion
The compactor is responsible for enforcing the storage retention, deleting the blocks that contain samples that are older than the configured retention period from the long-term storage.
The storage retention is disabled by default, and no data will be deleted from the long-term storage unless you explicitly configure the retention period.
```

File: docs/sources/operators-guide/architecture/components/compactor/index.md [hunk 2]
```diff
@@ -115,6 +115,13 @@ Soft deletion is based on a small `deletion-mark.json` file stored within the bl
...
+
+Refer to [configure metrics storage retention]({{< relref "../../../configure/configure-metrics-storage-retention.md" >}}) for more information.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/architecture/components/compactor/index.md:123]:
```suggestion
For more information, refer to [Configure metrics storage retention]({{< relref "../../../configure/configure-metrics-storage-retention.md" >}}).
```

File: docs/sources/operators-guide/configure/configure-metrics-storage-retention.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+menuTitle: "Configure metrics storage retention"
+description: "Learn how to configure Grafana Mimir metrics storage retention."
+weight: 70
+---
+
+# Configure Grafana Mimir metrics storage retention
+
+Grafana Mimir stores the metrics in a object storage.
+
+By default, metrics stored in the object storage are never deleted, and the storage utilization will increase over time.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/configure/configure-metrics-storage-retention.md:12]:
```suggestion
By default, metrics that are stored in the object storage are never deleted, and the storage utilization will increase over time.
```

File: docs/sources/operators-guide/configure/configure-metrics-storage-retention.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+You can configure the object storage retention to automatically delete all the metrics data older than the configured period.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/configure/configure-metrics-storage-retention.md:13]:
```suggestion
You can configure the object storage retention to automatically delete all of the metrics data older than the configured period.
```

File: docs/sources/operators-guide/configure/configure-metrics-storage-retention.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## How to configure the storage retention
+
+The [compactor]({{< relref "../architecture/components/compactor/index.md" >}}) is the Mimir component responsible to enforce the storage retention.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/configure/configure-metrics-storage-retention.md:17]:
```suggestion
The [compactor]({{< relref "../architecture/components/compactor/index.md" >}}) is the Mimir component that is responsible for enforcing the storage retention.
```

File: docs/sources/operators-guide/configure/configure-metrics-storage-retention.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+To configure the storage retention you can set the CLI flag `-compactor.blocks-retention-period` or change the following YAML configuration:
+
+```yaml
+limits:
+  # Delete from storage metrics data older than 1 year.
+  compactor_blocks_retention_period: 1y
+```
+
+You can configure the storage retention on a per-tenant basis settings overrides in the [runtime configuration]({{< relref "about-runtime-configuration.md" >}}):
```
Comment (osg-grafana) [inline docs/sources/operators-guide/configure/configure-metrics-storage-retention.md:26]:
```suggestion
To configure the storage retention on a per-tenant basis, set overrides in the [runtime configuration]({{< relref "about-runtime-configuration.md" >}}):
```
Please vet this change for accuracy. Jetlagged atm and had a hard time parsing the original sentence. :)

File: docs/sources/operators-guide/configure/configure-metrics-storage-retention.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    compactor_blocks_retention_period: 2y
+  tenant3:
+    # Disable retention for tenant3's metrics (never delete its data).
+    compactor_blocks_retention_period: 0
+```
+
+## Per-series retention
+
+Grafana Mimir doesn't support per-series deletion and retention.
+Prometheus' [Delete series API](https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series) is not supported by Mimir.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/configure/configure-metrics-storage-retention.md:44]:
```suggestion
Grafana Mimir doesn’t support per-series deletion and retention, nor does it support Prometheus' [Delete series API](https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series).
```

File: docs/sources/operators-guide/configure/configure-metrics-storage-retention.md (hunk 6 above)
Comment (pracucci) [reply docs/sources/operators-guide/configure/configure-metrics-storage-retention.md:26]: It's correct!

**Comment by pracucci** [comment]
Thanks @osg-grafana! I accepted all your suggestions 😉

**Comment by chiragg6** [comment]
@pracucci Is there any plan to have API to manage tenant-based retention periods?
I would be happy to contribute to the same.

**Pull request merged**